_posts/2020-11-05-springCloudCircuitBreaker.md

```diff
@@ -0,0 +1,550 @@
+```
+Console logging
+===============================
+Inside RestController method..
+Simulateretry is true, so try to simulate exception scenario.
+Random Number : 1
+ 
+===============================
+Inside RestController mathod..
+Simulateretry is true, so try to simulate exception scenario.
+Random Number : 2
+Simulateretry is true, so try to simulate exception scenario.
+Random Number : 2
+Simulateretry is true, so try to simulate exception scenario.
+Random Number : 0
+All retries completed, so Fallback method called!!!
+```
+
+Now try with 'http://localhost:8080/retry?simulateretry=true&simulateretryfallback=true', you will get fallback response after retry limit as from code everytime we are throwing RuntimeException. 
+Here are my last few lines of code, it tried 3 times before sending the response.
+```
+Console logging
+===============================
+Inside RestController method..
+Simulateretry is true, so try to simulate exception scenario.
+Simulateretry is true, so try to simulate exception scenario.
+Simulateretry is true, so try to simulate exception scenario.
+All retries completed, so Fallback method called!!!
+```
+
+Spring Retry provides declarative retry support for Spring applications. A subset of the project includes the ability to implement circuit breaker functionality. 
+Spring Retry provides a circuit breaker implementation via a combination of it’s CircuitBreakerRetryPolicy and a stateful retry. 
+All circuit breakers created using Spring Retry will be created using the CircuitBreakerRetryPolicy and a DefaultRetryState. 
+Both of these classes can be configured using SpringRetryConfigBuilder.
+To provide a default configuration for all of your circuit breakers create a Customize bean that is passed a SpringRetryCircuitBreakerFactory. 
+The configureDefault method can be used to provide a default configuration.
+
+
+```
+@Bean
+public Customizer<SpringRetryCircuitBreakerFactory> defaultCustomizer() {
+    return factory -> factory.configureDefault(id -> new SpringRetryConfigBuilder(id)
+        .retryPolicy(new TimeoutRetryPolicy()).build());
+}
+```
+
+Similarly to providing a default configuration, you can create a Customize bean this is passed a SpringRetryCircuitBreakerFactory.
+```
+@Bean
+public Customizer<SpringRetryCircuitBreakerFactory> slowCustomizer() {
+    return factory -> factory.configure(builder -> builder.retryPolicy(new SimpleRetryPolicy(1)).build(), "slow");
+}
+```
+In addition to configuring the circuit breaker that is created you can also customize the circuit breaker after it has been created but before it is returned to the caller. 
+To do this you can use the addRetryTemplateCustomizers method. This can be useful for adding event handlers to the RetryTemplate.
```

Comment:
```suggestion
To do this you can use the `addRetryTemplateCustomizers` method. This can be useful for adding event handlers to the `RetryTemplate`.
```